crates/brainfuck_vm/src/machine.rs

```diff
@@ -43,9 +56,13 @@ impl MachineBuilder {
     }
 
     /// Builds the [`Machine`] instance with the provided configuration.
-    pub fn build(self) -> Result<Machine, &'static str> {
+    pub fn build(self) -> Result<Machine, MachineError> {
         if self.input.is_none() || self.output.is_none() {
-            return Err("Input and output streams must be provided");
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "Input and output streams must be provided",
+            )
+            .into());
```

Comment:
Shouldn't it be `MachineError::IoError`  instead?

Reply:
Mhhh will be converted directly to this since there is a From. The message is only internal to the error but this is indeed an MachineError::IOError